Sources/Datadog/DDTracerConfiguration.swift

```diff
@@ -0,0 +1,32 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+extension DDTracer {
+    /// Datadog Tracer configuration.
+    public struct Configuration {
+        /// The service name that will appear in traces (if not provided or `nil`, the SDK default `serviceName` will be used).
+        public var serviceName: String?
+
+        /// Initializes the Datadog Tracer configuration.
+        /// - Parameter serviceName: the service name that will appear in traces (if not provided or `nil`, the SDK default `serviceName` will be used).
+        public init(serviceName: String? = nil) {
+            self.serviceName = serviceName
+        }
+    }
+
+    /// Datadog Tracer configuration resolved with defaults coming from the other source.
+    internal struct ResolvedConfiguration {
+        internal let serviceName: String
+
+        /// TODO: RUMM-409 Defaults should be resolved using `Datadog.instance`. I don't do it now, because the SDK-wide `serviceName`
+        ///  must be first done on `master` branch, then merged to `tracing` and update here.
+        init(tracerConfiguration: Configuration) {
+            self.serviceName = tracerConfiguration.serviceName ?? "ios"
+        }
+    }
```

Comment:
couldn't we use `DDTracer.Configuration` for this? why do we need another type?
(i forgot if we'd already discussed about that)

Reply:
The `DDTracer.Configuration` contains optional `serviceName: String?`. Here we resolve it to non-optional `serviceName: String` to cut-off optionality consideration in other SDK components.

However, this is likely to change after #102 (PR to `master`), so let's assume the `ResolvedConfiguration` will be removed in favour of `Datadog.ValidConfiguration` from #102. I think that when #102 is approved and merged, I'll create a separate, `master` → `tracing` PR to highlight this change and review it separately, as it will be non-trivial merge.